src/aleph_client/commands/instance.py

```diff
@@ -0,0 +1,219 @@
+            rootfs_size
+            or input(f"Size in MiB ? [{settings.DEFAULT_ROOTFS_SIZE}] ")
+            or settings.DEFAULT_ROOTFS_SIZE
+    )
+
+    rootfs_name = (
+            rootfs_name
+            or input(f"Root filesystem name ? [{settings.DEFAULT_ROOTFS_NAME}] ")
+            or settings.DEFAULT_ROOTFS_NAME
+    )
+
+    volumes = []
+
+    # Check if the volumes are empty
+    if persistent_volume is None or ephemeral_volume is None or immutable_volume is None:
+        for volume in prompt_for_volumes():
+            volumes.append(volume)
+            typer.echo("\n")
+
+    # else  Parse all the volumes that have passed as the cli parameters and put it into volume list
+    else:
+        if len(persistent_volume) > 0:
+            persistent_volume_dict = volume_to_dict(volume=persistent_volume)
+            volumes.append(persistent_volume_dict)
+        if len(ephemeral_volume) > 0:
+            ephemeral_volume_dict = volume_to_dict(volume=ephemeral_volume)
+            volumes.append(ephemeral_volume_dict)
+        if len(immutable_volume) > 0:
+            immutable_volume_dict = volume_to_dict(volume=immutable_volume)
+            volumes.append(immutable_volume_dict)
+
+    subscriptions: Optional[List[Dict]]
+    if beta and yes_no_input("Subscribe to messages ?", default=False):
+        content_raw = input_multiline()
+        try:
+            subscriptions = json.loads(content_raw)
+        except json.decoder.JSONDecodeError:
+            typer.echo("Not valid JSON")
+            raise typer.Exit(code=2)
+    else:
+        subscriptions = None
+
+    try:
+        # Register the instance
+        message, status = synchronous.create_instance(
+            account=account,
+            rootfs=rootfs,
+            rootfs_size=rootfs_size,
+            rootfs_name=rootfs_name,
+            storage_engine=StorageEnum.storage,
+            channel=channel,
+            memory=memory,
+            vcpus=vcpus,
+            timeout_seconds=timeout_seconds,
+            persistent=persistent,
+            volumes=volumes,
+            subscriptions=subscriptions,
+        )
+        logger.debug("Upload finished")
+        if print_messages or print_instance_message:
+            typer.echo(f"{message.json(indent=4)}")
+
+        hash: str = message.item_hash
+        hash_base32 = b32encode(b16decode(hash.upper())).strip(b"=").lower().decode()
+
+        typer.echo(
+            f"Your instance has been deployed on Aleph .\n\n"
+            "Available on:\n"
+            f"  {settings.VM_URL_PATH.format(hash=hash)}\n"
+            f"  {settings.VM_URL_HOST.format(hash_base32=hash_base32)}\n"
+            "Visualise on:\n  https://explorer.aleph.im/address/"
+            f"{message.chain}/{message.sender}/message/INSTANCE/{hash}\n"
+        )
+
+    finally:
+        # Prevent aiohttp unclosed connector warning
+        asyncio.run(get_fallback_session().close())
+
+
+@app.command()
+def unpersist(
```

Comment:
Not really applicable to instances. That's a whole other issue, but we should improve the communication between CCNs and CRNs now that CCNs are more reliable. Having a P2P pubsub channel or a websocket to retrieve operations on programs and instances would make it possible to delete a VM once the instance message is forgotten, for example.

In the meantime: remove this command as it does not make sense to "unpersist" a VM instance.

@hoh do we want a `delete` command that just forgets the instance instead?

Reply:
I have used the same existing method on the VMs, but we can change it to delete, that also for me have better sense.